plugins/modules/smart_proxy.py

```diff
@@ -101,13 +110,50 @@ def main():
         foreman_spec=dict(
             name=dict(required=True),
             url=dict(required=True),
+            lifecycle_environments=dict(required=False, type='entity_list'),
             download_policy=dict(required=False, choices=['background', 'immediate', 'on_demand']),
         ),
-        required_plugins=[('katello', ['download_policy'])],
+        required_plugins=[('katello', ['lifecycle_environments', 'download_policy'])],
     )
 
     with module.api_connection():
-        module.run()
+        handle_lifecycle_environments = not module.desired_absent and 'lifecycle_environments' in module.foreman_params
+        if handle_lifecycle_environments:
+            module.lookup_entity('lifecycle_environments')
+
+        lifecycle_environments = module.foreman_params.pop('lifecycle_environments', None)
+        smart_proxy = module.run()
+
+        payload = {
+            'id': smart_proxy['id'],
+        }
+        current_lces = module.resource_action('capsule_content', 'lifecycle_environments', payload, record_change=False)
```

Comment:
This needs to be guarded by a `if handle_lifecycle_environments` as otherwise this is also executed on non-Katello installations and fails. (so just move the payload and current lines behind the `if` in the next line?)

Reply:
Makes sense, fixed!